Replaced Boolean objects with primitive boolean

siavash1986 opened the pull request:
Replaced `Boolean` objects with primitive booleans where instantiating Objects were not necessary. Booleans are more expensive to create, have 3 states and therefore increase the risk of NullPointerException

#### What has been done to verify that this works as intended?
ran lint, checkstyle and PMD

#### Why is this the best possible solution? Were any other approaches considered?

#### Are there any risks to merging this code? If so, what are they?
It's safe

siavash1986 committed:
Replaced Boolean objects with primitive boolean whenever using Boolean objects were not necessary.

dcbriccetti commented on collect_app/src/main/java/org/odk/collect/android/activities/GeoPointMapActivity.java:
```diff
@@ -95,11 +95,11 @@
-    private Boolean draggable = false;
-    private Boolean intentDraggable = false;
-    private Boolean locationFromIntent = false;
+    private boolean foundFirstLocation = false;
```
boolean fields default to false, so I suggest removing “ = false”. This will reduce the code volume, and, I argue, improve readability.

siavash1986 replied:
That's right, I'll modify this code.
Thankyou @dcbriccetti

siavash1986 committed:
replaced File.delete() with FileUtils.deleteAndReport().
UtilFiles.deleteAndReport() gives a file a second chance to delete by calling File.deleteOnExit() and report the result, It's a bad practice to ignore return values.
Also this commit will log the return value of File.renameTo() and File.mkdirs() if they fail

siavash1986 commented:
sorry for the wrong second commit message, it will be fixed in the next push.
The second commit removes initializing primitive booleans to false since primitive booleans are defaulted to false and initializing them once again is redundant.
